### PR TITLE
Add `styled-components-a11y` to eslint configuration

### PR DESCRIPTION
**Pull request by SudharakaP**
This adds the accessibility plugin styled-components-a11y to our eslint configuration. ~~I think there's a bug in this package when used with `styled-component` attributes. This is created to illustrate that. The relevant issue I create is, https://github.com/brendanmorrell/eslint-plugin-styled-components-a11y/issues/15~~

Resolves https://github.com/opencollective/opencollective/issues/3980

### CODE REVIEW

**Comment by brendanmorrell** [comment]
@SudharakaP I responded [in your issue](https://github.com/opencollective/opencollective/issues/3980), but you are the first to report this issue, so thank you. I'll take a look asap and publish a patch so you can add the plugin to this repo. I'll let you know when the fix version is up. Thanks!

File: .eslintrc.json
Comment (Betree) [inline .eslintrc.json:80]: I find that using "warn" instead of `1` is clearer, especially because doesn't require any specific knowledge about ESLint to understand it. It's also the pattern we use above, so I think we should improve the consistency here.

**Review by Betree — COMMENTED** [review]
No blocker for me.

@znarf wdyt?

File: .eslintrc.json
Comment (SudharakaP) [reply .eslintrc.json]: @Betree : Thanks. I've changed that to `warn` for consistence. I think now it looks better. 👍🏽

**Comment by SudharakaP** [comment]
@Betree : I think this is good to merge if everything is ship shape on the review side. If there's anything else to be done let me know. I'll do the accessibility fixes on separate PRs if you don't mind, just so that this is not bloated with lot of changes. 😄

**Review by znarf — APPROVED** [review]
Good for me!

**Review by Betree — APPROVED** [review]
We need a small rebase for the package-lock conflict, after that, I'm all good to merge

**Comment by SudharakaP** [comment]
> We need a small rebase for the package-lock conflict, after that, I'm all good to merge

@Betree : Updated `package-lock.json`. Hope everything is good now. 😄

**Pull request merged**
